### ISSUE TITLE
Questions about evaluate time

### ISSUE THREAD

**Issue by Yuan0320** [AGI-Edgerunners/LLM-Adapters#40]
Hi @HZQ950419, thanks for your great work! Here I wonder how long your evaluation phase took? I used a single V100, and the evaluation phase seems a bit time consuming, e.g. I spent about 4 hours on the AddSub data set. Is this normal?

**Comment by HZQ950419** [comment]
Hi,

Yes, the evaluation will take a very long time. The gsm8k will take around 20+ hours for V100 32G without load_8bit. 
Please let us know if you have further questions!

**Comment by Yuan0320** [comment]
Thanks for the response! More importantly, I ran the experiments with different GPU and found that the evaluation accuracy on the fine-tuned model trained with V100 was significantly worse. I wonder if you have ever experienced this? I'm not sure if I'm going wrong somewhere.

| GPU | eval loss | AddSub Acc | 
|------|------------|--------------------|
| / | / | 78.5 (paper reported) | 
|Quadro RTX 6000 | 0.5750 | 86.07 | 
| GeForce RTX 3090 | 0.5718 | 86.58 |
| Tesla V100 | 0.6179 | 60.99 | 

 PS: I trained on `math_data.json` and tested on `AddSub`. All experiments were conducted in the same env.

**Comment by HZQ950419** [comment]
Hi,

please use the following command to reproduce the result for LLaMA-7B-LoRA.

`CUDA_VISIBLE_DEVICES=0 python finetune.py   --base_model 'yahma/llama-7b-hf'   --data_path 'math_10k.json'   --output_dir './trained_models/llama-7b-lora-math/'   --batch_size 16  --micro_batch_size 4   --num_epochs 3   --learning_rate 3e-4   --cutoff_len 256   --val_set_size 120 --eval_step 80 --save_step 80  --adapter_name lora --target_modules '["q_proj", "k_proj", "v_proj", "up_proj", "down_proj"]' --lora_r 32 --lora_alpha 64`

**Comment by Yuan0320** [comment]
Thanks! This helps me a lot. Currently the training dataset, e.g., `math_data.json`, covers all six math reasoning datasets, I wonder that can we get the training data for each datasets? Or other ways to get the dataset to which each sample in the training data belongs.
